src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -40,6 +40,15 @@ public SNIPacket(int headerSize, int dataSize)
         /// </summary>
         public bool IsInvalid => _data is null;
 
+        public bool HasCompletionCallback => !(_completionCallback is null);
+
+        /// <summary>
+        /// Packet data
+        /// </summary>
+        public void Dispose()
```

Comment:
Nit: this can be `public void Dispose() => Release(); ` followed by a new line.

Reply:
Done. Dispose should be going away in a future PR because with full packet recycling you never want to dispose them you just call Release() to put them back into the pool.